pkg/nsx/services/inventory/pod.go

```diff
@@ -82,3 +91,15 @@ func (s *InventoryService) DeleteContainerApplicationInstance(externalId string,
 	}
 	return nil
 }
+
+func (s *InventoryService) DeleteStalePods() error {
```

Comment:
I think we don't need this, isn't CleanStaleInventoryApplicationInstance enough?

Reply:
DeleteStalePods is specific for stale pod which doesn't have project_id (antrea have this scenario). 
		if applicationInstance.ContainerProjectId == "" {
				// ContainerProjectId is the index for applicationInstanceStore and cannot be empty.
				// The case should not happen and we need to put this one into stale cache and clean it during resync if it did happen.
CleanStaleInventoryApplicationInstance only for normal GC case.